src/sagemaker_xgboost_container/data_utils.py

```diff
@@ -300,15 +300,17 @@ def _get_csv_dmatrix_file_mode(files_path, csv_weights):
     return dmatrix
 
 
-def _get_csv_dmatrix_pipe_mode(pipe_path, csv_weights):
+def _get_csv_dmatrix_pipe_mode(pipe_path, csv_weights, subsample_on_read):
     """Get Data Matrix from CSV data in pipe mode.
 
     :param pipe_path: SageMaker pipe path where CSV formatted training data is piped
     :param csv_weights: 1 if instance weights are in second column of CSV data; else 0
+    :param subsample_on_read: None or a value in (0, 1] to indicate how much of the dataset should
+            be read into memory.
     :return: xgb.DMatrix or None
     """
     try:
-        dataset = [mlio.SageMakerPipe(pipe_path)]
+        dataset = [mlio.SageMakerPipe(pipe_path, fifo_id=0)]
```

Comment:
Does the fifo_id not default to 0?

Reply:
Yes, it does. The reason I had to explicitly specify it is because we have unit tests for pipe mode that all run in the same process. Without this parameter secondary unit tests fail because they look for FIFO 1, but the pipe mode emulator gets restarted with each unit test. Ideally those tests should not be unit but integration tests in the future. Explicitly specifying a `fifo_id` of `0` does not change the behavior, but it is superfluous